jablib/src/main/java/org/jabref/logic/importer/fileformat/pdf/CitationsFromPdf.java

```diff
@@ -0,0 +1,39 @@
+package org.jabref.logic.importer.fileformat.pdf;
+
+import java.nio.file.Path;
+
+import org.jabref.logic.ai.AiService;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.importer.plaincitation.LlmPlainCitationParser;
+import org.jabref.logic.preferences.JabRefCliPreferences;
+import org.jabref.logic.util.CurrentThreadTaskExecutor;
+import org.jabref.logic.util.NotificationService;
+
+/// Wrapper around the different options to extract citations from a PDF
+///
+/// We deliberately opt for passing whole [JabRefCliPreferences] to ease calling this helper methods
+public class CitationsFromPdf {
+
+    public static ParserResult extractCitationsUsingTextMatching(JabRefCliPreferences preferences, Path path) {
```

Comment:
```suggestion
    public static ParserResult extractCitationsUsingRuleBasedAlgorithm(JabRefCliPreferences preferences, Path path) {
```

I think `Rule based` is too wordly, but I took it from AI history. For consistency, let's use it for now